ta/pkcs11/src/pkcs11_token.c

```diff
@@ -43,6 +45,51 @@ unsigned int get_token_id(struct ck_token *token)
 	return id;
 }
 
+struct pkcs11_client *tee_session2client(uintptr_t tee_session)
```

Comment:
Why the `uintptr_t`?
You could use `struct pkcs11_client` here instead and it might be a bit easier to follow the code. `struct pkcs11_client` doesn't have to be exposed in the .h file if you'd rather keep it opaque.

Reply:
> Why the `uintptr_t`?

Could be `void *` instead of `uintptr_t`.

> You could use struct pkcs11_client here (...)

This helper converts TEE session reference (which is a `void *` by the spec) into a `struct pkcs11_client` reference. So keep `void *` as input argument is simpler I guess. I could straight change InvokeCommand handler as:
```patch
 TEE_Result TA_InvokeCommandEntryPoint(void *tee_session, uint32_t cmd,
 				      uint32_t ptypes,
 				      TEE_Param params[TEE_NUM_PARAMS])
 {
-	uintptr_t teesess = (uintptr_t)tee_session;
+	struct pkcs11_client *client = tee_session2client(tee_session);
 	uint32_t rc = 0;
```

> `struct pkcs11_client` doesn't have to be exposed in the .h if you'd rather keep it opaque.

Right. I've move the definition into this .c file.